src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -428,6 +467,99 @@ void AccessControlAttribute::OnEntryChanged(const SubjectDescriptor * subjectDes
     ChipLogError(DataManagement, "AccessControlCluster: event failed %" CHIP_ERROR_FORMAT, err.Format());
 }
 
+#if CHIP_CONFIG_USE_ACCESS_RESTRICTIONS
+CHIP_ERROR AccessControlAttribute::ReadCommissioningArl(AttributeValueEncoder & aEncoder)
+{
+    auto accessRestrictionProvider = GetAccessControl().GetAccessRestrictionProvider();
+    if (accessRestrictionProvider == nullptr)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
```

Comment:
Please return CHIP_NO_ERROR here, which will leave the list empty (no restrictions). It is legal to support restrictions but not have started using a provider yet.

Reply:
Same comment for `ReadArl` below

Reply:
I had to call EncodeList and let it check for presence of the restriction provider, otherwise I got a bad response.